pkg/manager/cli_download_install.go

```diff
@@ -0,0 +1,319 @@
+	_, err = controllerutil.CreateOrUpdate(context.TODO(), hubclient, cliDownload, func() error { return nil })
+	if err != nil {
+		log.Error(err, "failed to create or update hypershift-cli-download ConsoleCLIDownload")
+		return err
+	}
+	log.Info("hypershift-cli-download ConsoleCLIDownload was applied successfully")
+
+	return nil
+}
+
+func getCLIDeployment(cliImage string, envVars []corev1.EnvVar, log logr.Logger) (*appsv1.Deployment, error) {
+	depFile, err := fs.ReadFile("manifests/cli/deployment.yaml")
+	if err != nil {
+		log.Error(err, "failed to read manifests/cli/deployment.yaml")
+		return nil, err
+	}
+
+	dep := &appsv1.Deployment{}
+	err = yaml.Unmarshal(depFile, &dep)
+	if err != nil {
+		log.Error(err, "failed to parse manifests/cli/deployment.yaml")
+		return nil, err
+	}
+
+	// set the deployment with the hypershift_cli image from CSV
+	dep.Spec.Template.Spec.Containers[0].Image = cliImage
+
+	// set proxy environment variables if exist in the addon manager deployment
+	containerEnvVars := []corev1.EnvVar{}
+	for _, envVar := range envVars {
+		if strings.HasSuffix(envVar.Name, "_PROXY") {
+			containerEnvVars = append(containerEnvVars, corev1.EnvVar{Name: envVar.Name, Value: envVar.Value})
+		}
+	}
+
+	if len(containerEnvVars) > 0 {
+		dep.Spec.Template.Spec.Containers[0].Env = containerEnvVars
+	}
+
+	return dep, nil
+}
+
+func getService(log logr.Logger) (*corev1.Service, error) {
+	serviceFile, err := fs.ReadFile("manifests/cli/service.yaml")
+	if err != nil {
+		log.Error(err, "failed to read manifests/cli/service.yaml")
+		return nil, err
+	}
+
+	service := &corev1.Service{}
+	err = yaml.Unmarshal(serviceFile, &service)
+	if err != nil {
+		log.Error(err, "failed to parse manifests/cli/service.yaml")
+		return nil, err
+	}
+
+	return service, nil
+}
+
+func getRoute(log logr.Logger) (*routev1.Route, error) {
+	routeFile, err := fs.ReadFile("manifests/cli/route.yaml")
+	if err != nil {
+		log.Error(err, "failed to read manifests/cli/route.yaml")
+		return nil, err
+	}
+
+	route := &routev1.Route{}
+	err = yaml.Unmarshal(routeFile, &route)
+	if err != nil {
+		log.Error(err, "failed to parse manifests/cli/route.yaml")
+		return nil, err
+	}
+
+	return route, nil
+}
+
+func getConsoleDownload(routeUrl string, log logr.Logger) (*consolev1.ConsoleCLIDownload, error) {
+	log.Info("using route URL: " + routeUrl)
+	cliDownloadFile, err := fs.ReadFile("manifests/cli/consoledownload.yaml")
+	if err != nil {
+		log.Error(err, "failed to read manifests/cli/consoledownload.yaml")
+		return nil, err
+	}
+
+	cliDownload := &consolev1.ConsoleCLIDownload{}
+	err = yaml.Unmarshal(cliDownloadFile, &cliDownload)
+	if err != nil {
+		log.Error(err, "failed to parse manifests/cli/consoledownload.yaml")
+		return nil, err
+	}
+
+	links := []consolev1.CLIDownloadLink{
+		{
+			Href: "https://" + routeUrl + "/hypershift-linux-amd64.tar.gz",
+			Text: "Download oc hcp for Linux for x86_64",
```

Comment:
If we change it in the docs to Hosted control plane CLI, maybe we should change it here as well.

Reply:
`oc hcp` is the CLI tool like `oc`.